src/hotspot/share/runtime/objectMonitor.inline.hpp

```diff
@@ -56,6 +56,11 @@ inline int ObjectMonitor::waiters() const {
   return _waiters;
 }
 
+inline bool ObjectMonitor::has_owner() const {
+  void* owner = owner_raw();
+  return owner != NULL && owner != DEFLATER_MARKER;
```

Comment:
You could also do:
```
    return owner() != nullptr;
```
and take advantage of the fact that `owner()` filters out DEFLATER_MARKER for you.